splitio/client/factory.py

```diff
@@ -317,6 +325,14 @@ def _build_in_memory_factory(api_key, cfg, sdk_url=None, events_url=None,  # pyl
         'impressions': InMemoryImpressionStorage(cfg['impressionsQueueSize']),
         'events': InMemoryEventStorage(cfg['eventsQueueSize']),
     }
+    imp_counter = ImpressionsCounter() if cfg['impressionsMode'] != ImpressionsMode.DEBUG else None
+
+    strategies = {
```

Comment:
we should instantiate only the required one

Reply:
The imp_counter is required for optimized and none. Not sure what is the issue here?